src/components/ModalDetailsMovie/ModalDetailsMovie.js

```diff
@@ -0,0 +1,215 @@
+import Loading from "../Loading/Loading";
+
+class ModalDetailsMovie extends Component {
+  state = {
+    isLoading: true,
+    error: null,
+    movie: {},
+    onWatchlist: false,
+    typeAction: "add",
+    message: "",
+  };
+
+  componentDidMount() {
+    this.fetchMovie();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const isFavMovie = Object.values(this.context.favMovies)
+      .flat()
+      .find((movie) => movie.id === this.props.movieID);
+
+    if (!!isFavMovie !== prevState.onWatchlist) {
+      let tempTypeAction;
+      if (prevState.typeAction === "add") tempTypeAction = "remove";
+      else if (prevState.typeAction === "remove") tempTypeAction = "add";
+      this.setState({
+        onWatchlist: !prevState.onWatchlist,
+        typeAction: tempTypeAction,
+        message: this.context.message,
+      });
+    }
+    if (this.context.reload) {
+      this.context.handleOffReload();
+      this.fetchMovie();
+    }
+  }
+
+  fetchMovie = () => {
+    const apiKey = "879b3e71";
+    const movieID = this.props.movieID;
+    const API = `https://www.omdbapi.com/?i=${movieID}&apikey=${apiKey}`;
+
+    fetch(API)
+      .then(
+        (response) => {
+          if (response.ok) {
+            return response;
+          } else {
+            const error = new Error(
+              `Error ${response.status} ${response.statusText}. Please try agine ...`
+            );
+            error.response = response;
+            throw error;
+          }
+        },
+
+        (error) => {
+          const errMessage = new Error(
+            `${error.message}. Please check your network connection and try agine later.`
+          );
+          throw errMessage;
+        }
+      )
+      .then((response) => response.json())
+      .then((response) => {
+        if (response.Response === "True") {
+          this.setState({
+            isLoading: false,
+            movie: { ...response },
+          });
+        } else {
+          const errMessage = new Error(response.Error);
+          throw errMessage;
+        }
+      })
+      .catch((error) =>
+        this.setState({
+          isLoading: false,
+          error: error.message,
+        })
+      );
+
+    const isFavMovie = Object.values(this.context.favMovies)
+      .flat()
+      .find((movie) => movie.id === this.props.movieID);
+
+    if (!!isFavMovie)
+      this.setState({ onWatchlist: true, typeAction: "remove" });
+    else if (!isFavMovie)
+      this.setState({ onWatchlist: false, typeAction: "add" });
+  };
+
+  render() {
+    const { closeModal, movieID } = this.props;
+    const {
+      Title,
+      Year,
+      Runtime,
+      Genre,
+      Actors,
+      Plot,
+      Awards,
+      Poster,
+      imdbRating,
+      imdbVotes,
+    } = this.state.movie;
+
+    let listActors;
+    if (Actors) {
+      listActors = Actors.split(",").map((actor, index) => (
+        <li key={index}>{actor}</li>
+      ));
+    }
```

Comment:
to powinien być osobny komponent poza tym renderem, który przyjmuje jako props `Actors`